masonry/src/widget/variable_label.rs

```diff
@@ -0,0 +1,393 @@
+// Copyright 2019 the Xilem Authors and the Druid Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! A label with support for animated variable font properties
+
+use accesskit::Role;
+use kurbo::{Affine, Point, Size};
+use parley::fontique::Weight;
+use parley::layout::Alignment;
+use parley::style::{FontFamily, FontStack};
+use smallvec::SmallVec;
+use tracing::{trace, trace_span, Span};
+use vello::peniko::BlendMode;
+use vello::Scene;
+
+use crate::text::{TextBrush, TextLayout, TextStorage};
+use crate::widget::WidgetMut;
+use crate::{
+    AccessCtx, AccessEvent, ArcStr, BoxConstraints, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx,
+    PaintCtx, PointerEvent, StatusChange, TextEvent, Widget, WidgetId,
+};
+
+use super::LineBreaking;
+
+// added padding between the edges of the widget and the text.
+pub(super) const LABEL_X_PADDING: f64 = 2.0;
+
+/// An `f32` value which can move towards a target value at a linear rate over time.
+#[derive(Clone, Debug)]
+pub struct AnimatedF32 {
+    /// The value which self will eventually reach.
+    target: f32,
+    /// The current value
+    value: f32,
+    // TODO: Provide different easing functions, instead of just linear
+    /// The change in value every millisecond, which will not change over the lifetime of the value.
+    rate_per_millisecond: f32,
+}
+
+impl AnimatedF32 {
+    /// Create a value which is not changing.
+    pub fn stable(value: f32) -> Self {
+        assert!(value.is_finite());
+        AnimatedF32 {
+            target: value,
+            value,
+            rate_per_millisecond: 0.,
+        }
+    }
+
+    /// Move this value to the `target` over `target` milliseconds.
+    pub fn move_to(&mut self, target: f32, over_millis: f32) {
+        assert!(target.is_finite());
+        self.target = target;
+        self.rate_per_millisecond = (self.target - self.value) / over_millis;
```

Comment:
I think we should either add a division by zero guard here (epsilon?), or warn the user somewhere (with a doc-comment). (I lean towards defusing this)

Reply:
Is that not achieved by the very next statement? I do agree that this should be documented, good point

Reply:
I could produce a crash by using `0.0` in the xilem example

Reply:
Oh I see. Yeah, if `over_millis` is zero, it makes sense to just set the value directly. I need to document that possibility properly.